privacyidea/static/components/token/controllers/tokenControllers.js

```diff
@@ -280,11 +281,14 @@ myApp.controller("tokenEnrollController", function ($scope, TokenFactory,
             $scope.form.hashlib = $scope.systemDefault['totp.hashlib'] || 'sha1';
             $scope.form.timeStep = parseInt($scope.systemDefault['totp.timeStep'] || '30');
         }
-        if ($scope.form.type === "vasco") {
+        if (["vasco", "indexedsecret"].indexOf($scope.form.type) >= 0) {
             $scope.form.genkey = false;
```

Comment:
We should move this into the `preset_indexedsecret()` function since this function only gets called when the tokentype gets changed.
If a user is only allowed to enroll IndexedSecretTokens and enters a shared secret, the token creation fails because `genkey` is still true.

Reply:
done in 352850f8